java/main/com/google/privacy/differentialprivacy/BoundedMean.java

```diff
@@ -175,6 +175,74 @@ public double computeResult() {
     return clamp(normalizedNoisedSum / noisedCount + midpoint);
   }
 
+  /**
+   * Computes a confidence interval that contains the true mean with a probability greater or equal
+   * to {@code 1 - alpha}. The computation is based exclusively on the noised data and the privacy
+   * parameters. Thus no privacy budget is consumed by this operation.
+   *
+   * <p>See <a href="https://github.com/google/differential-privacy/tree/main/common_docs/confidence_intervals.md">
+   * the confidence intervals doc</a>.
+   */
+  public ConfidenceInterval computeConfidenceInterval(double alpha) {
+    if (!resultReturned) {
+      throw new IllegalStateException("computeResult() must be called before calling computeConfidenceInterval()");
+    }
+    // The confidence interval of bounded mean is derived from confidence intervals of mean's numerator and denominator.
+    // The respective confidence levels 1 - alphaNum and 1 - alphaDen can be chosen arbitrarily as long as (1 - alphaNum) *
+    // (1 - alphaDen) = 1 - alpha. The following is a brute force search for alphaNum that minimizes the size of the confidence
+    // interval of bounded mean.
+    double minSize = Double.POSITIVE_INFINITY;
+    ConfidenceInterval tightestConfInt = null;
+    for (int i = 1; i < 1000; i++) {
+      double alphaNum = (i / 1000.0) * alpha;
+      ConfidenceInterval confInt = computeConfidenceInterval(alpha, alphaNum);
+      double size = confInt.upperBound() - confInt.lowerBound();
+      if (size < minSize) {
+        minSize = size;
+        tightestConfInt = confInt;
+      }
+    }
+    return tightestConfInt;
+  }
+
+  /**
+   * See {@link #computeConfidenceInterval(double)}.
+   *
+   * <p>As opposed to the former function, this function computes a confidence interval for mean
+   * with the additional constraint that the confidence level of the mean's numerator is {@code 1 -
+   * alphaNum}.
+   */
+  ConfidenceInterval computeConfidenceInterval(double alpha, double alphaNum) {
```

Comment:
why package private?
Can't it be private? If it can then we should fix this, since the good practice is to use as low access level as possible

Reply:
It needs to be package private for testing. I also think it is a useful function to have it on package level.

Reply:
why fo you think it is useful outside of the scope of tests)? We have our public method, this method represent technicality of how we calculate these intervals, and shouldn't be important for other "users" (clients of the library or the library itself) because this algorithm might change in the future (but the public API will remain the same).

So since we need it for the test mark it please @VisibleForTesting

Reply:
As we discussed offline, We agreed to keep it package private with @**VisibleForTesting** annotation since it is used in testing. There is also a certain use case that might be reasonable for Variance yet not the main reason it is package private.